src/sage/modules/free_module_pseudomorphism.py

```diff
@@ -0,0 +1,443 @@
+            (z^2 + 6*z + 2, z^2 + 2*z + 1, 2*z^2 + 4*z)
+        """
+        D = self.domain()
+        C = self.codomain()
+        if D.is_ambient():
+            x = x.element()
+        else:
+            x = D.coordinate_vector(x)
+        if self._morphism is None:
+            x_twist = x
+        else:
+            x_twist = D(list(map(self._morphism, x)))
+        v = x_twist * self._matrix
+        if C.is_ambient():
+            v = C(v.list())
+        else:
+            v = C.linear_combination_of_basis(v)
+        if self._derivation is not None:
+            v += D(list(map(self._derivation, x)))
+        return v
+
+    def _repr_(self):
+        r"""
+        Return the string representation of a pseudomorphism.
+
+        TESTS::
+
+            sage: Fq.<z> = GF(7^3)
+            sage: M = Fq^3
+            sage: Frob = Fq.frobenius_endomorphism()
+
+            sage: f = M.pseudohom([[1,1,1], [2,2,2], [3,3,3]], Frob)
+            sage: f  # indirect doctest
+            Free module pseudomorphism (twisted by z |--> z^7) defined by the matrix
+            [1 1 1]
+            [2 2 2]
+            [3 3 3]
+            Domain: Vector space of dimension 3 over Finite Field in z of size 7^3
+            Codomain: Vector space of dimension 3 over Finite Field in z of size 7^3
+
+            sage: g = M.pseudohom([[1,1,1], [2,2,2], [3,3,3]], Frob, side="right")
+            sage: g  # indirect doctest
+            Free module pseudomorphism (twisted by z |--> z^7) defined as left-multiplication by the matrix
+            [1 1 1]
+            [2 2 2]
+            [3 3 3]
+            Domain: Vector space of dimension 3 over Finite Field in z of size 7^3
+            Codomain: Vector space of dimension 3 over Finite Field in z of size 7^3
+        """
+        twist = self.parent()._ore._repr_twist()
+        s = "Free module pseudomorphism (%s) defined " % twist
+        if self._side == "right":
+            s += "as left-multiplication "
+        s += "by the matrix\n%s\n" % self.matrix()
+        s += "Domain: %s\n" % self.domain()
+        s += "Codomain: %s" % self.codomain()
+        return s
+
+    def matrix(self):
+        r"""
+        Return the underlying matrix of this pseudomorphism.
+
+        It is defined as the matrix `M` whose lines (resp. columns if
+        ``side`` is ``right``) are the coordinates of the images of
+        the distinguished basis of the domain.
+
+        EXAMPLES::
+
+            sage: Fq.<z> = GF(7^3)
+            sage: Frob = Fq.frobenius_endomorphism()
+            sage: M = Fq^3
+            sage: f = M.pseudohom([[1, z, 3], [0, 1, z^2], [z+1, 1, 1]], Frob)
+            sage: f.matrix()
+            [    1     z     3]
+            [    0     1   z^2]
+            [z + 1     1     1]
+
+            sage: e1, e2, e3 = M.basis()
+            sage: f(e1)
+            (1, z, 3)
+            sage: f(e2)
+            (0, 1, z^2)
+            sage: f(e3)
+            (z + 1, 1, 1)
+
```

Comment:
To facilitate live examples (meaning running examples in live doc), explicit separation of examples is better. So I suggest 
```
        EXAMPLES::
            sage: Fq.<z> = GF(7^3)
            sage: Frob = Fq.frobenius_endomorphism()
            sage: M = Fq^3
            sage: f = M.pseudohom([[1, z, 3], [0, 1, z^2], [z+1, 1, 1]], Frob)
            sage: f.matrix()
            [    1     z     3]
            [    0     1   z^2]
            [z + 1     1     1]

        ::

            sage: e1, e2, e3 = M.basis()
```